### DISCUSSION TITLE
Permission bits for '/Users/expo/.netrc' should be 0600, but are 644

### DISCUSSION THREAD

**Discussion by Raamzeez** [expo/expo#19152] [Archived: Questions] [answered]
I am having a bit of an issue with my .netrc file when building my react native expo project (bare workflow) with eas build in production mode (not locally). I am running my project with this command `eas build --platform ios --profile production`. My package.json has `"eas-build-pre-install": "echo machine api.mapbox.com login mapbox password \"************************************\" >~/.netrc"
` in the "scripts" object in order to create the .netrc file on Expo's servers with the appropriate secret key data (censored). I have attached an image of the error that I encounter when trying to build the project: 

<img width="1106" alt="Screen Shot 2022-09-16 at 10 04 45 PM" src="https://user-images.githubusercontent.com/37488506/190836264-92cb8246-c049-47b4-a66b-ca0082f9f548.png">

The interesting thing is that building the project works on my local machine (using the `--local` flag). I believe this is because I ran the `chmod -R 0600 ./netrc` in my home directory. However, I don't know how to do this when building on Expo's servers. Do I include this command in my `eas-build-pre-install` script? If so, how do I do this? Thanks for any help. 

My previous issue regarding .netrc: [Previous GitHub Issue]( https://github.com/expo/expo/issues/17182) helped by @wodin

**Comment by wodin** [comment, marked as answer]
Try changing it to the following:

`"eas-build-pre-install": "umask 0077; echo machine api.mapbox.com login mapbox password \"************************************\" >~/.netrc"`

EDIT:

If the above doesn't work for some reason; do the following:

`"eas-build-pre-install": "echo machine api.mapbox.com login mapbox password \"************************************\" >~/.netrc; chmod 600 ~/.netrc"`

The former is the "safer" way to do it, since the file will be created with the correct permissions up front instead of briefly having the wrong permissions. Whether this difference matters in practice is another story.

**Reply by Raamzeez** [reply]
Sorry for the late response, didn't realize that I never marked this as the answer. The first or "safer" solution worked for me. Thanks for the help again!
